src/test/java/com/epam/edumanagementsystem/admin/rest/repository/AcademicCourseRepositoryTest.java

```diff
@@ -0,0 +1,37 @@
+package com.epam.edumanagementsystem.admin.rest.repository;
+
+import com.epam.edumanagementsystem.admin.model.entity.AcademicCourse;
+import com.epam.edumanagementsystem.config.PostgresTestContainerConfig;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.jdbc.Sql;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.test.context.jdbc.Sql.ExecutionPhase.AFTER_TEST_METHOD;
+import static org.springframework.test.context.jdbc.Sql.ExecutionPhase.BEFORE_TEST_METHOD;
+
+@SpringBootTest
```

Comment:
There is no need in testing JPA's derived query methods at all, since it will mean testing Spring Data JPA. It's advised to test only methods with native queries, because neither Hibernate nor Spring Data validate native queries at startup. P.S Anyway, it's a better approach to use sliced testing approach while testing repository methods, which means using single @DataJpaTest would fit perfectly here and instead of using test containers and @Sql or any other approach, you could just add @AutoConfigureTestDatabase(replace = none) to connect to the real DB. Only in this way you can completely be sure that the method won't fail in production.